skimage/measure/fit.py

```diff
@@ -662,9 +662,12 @@ def _dynamic_max_trials(n_inliers, n_samples, min_samples, probability):
     if n_inliers == 0:
         return np.inf
     inlier_ratio = n_inliers / n_samples
-    nom = max(_EPSILON, 1 - probability)
+    nom = 1 - probability
     denom = 1 - inlier_ratio**min_samples
-    # Avoid log(1) below turning into -inf
+    # Keep (de-)nominator in the range of (0, 1) so that it is always
```

Comment:
Isn't the range eps to 1?

Misspelled guaranteed

Reply:
The only reason I'm saying it here is because the work we're doing now is explicitly to *avoid* the interval of the denumerator being 0-1. Not sure why we need to limit the numerator. We could always say a minimum of 1 (or N) trials.

Either way, for the denumerator, at least, it feels important that we explicitly exclude 0 from that interval. Which you did, but which I misread. Perhaps I would have noticed if it was written `(0, 1]` and, again, not sure whether it's better to exclude 1 from the bracket.

Reply:
> Not sure why we need to limit the numerator. We could always say a minimum of 1 (or N) trials.

Without limiting the numerator we get a singed zero `-0` for the number of trials. Which seems strange but is technically speaking not wrong because it is equal to zero.

Since anyway we limit the denominator it seemed straight forward to do the same for the numerator due to a symmetry argument and we do not have the `-0` as the number of trials . So as a result of the discussion: https://github.com/scikit-image/scikit-image/pull/7065#discussion_r1716807273, I opened this PR.

But you are right, we can also limit the number of minimum trials. It is a matter of taste.